### ISSUE TITLE
Saml2HttpRedirect.Serialize() pads a lot of zeros to the end

### ISSUE THREAD

**Issue by justrand** [Innofactor/Innofactor.SuomiFiIdentificationClient#10]
I ran into a problem where the redirect URL to suomi.fi would have a lot of A/ 0:s at the end. After investigating the issue, I noticed that the Serialize function in Saml2HttpRedirect produces the extra As at the end. I tested the function with the proper XML output from suomi.fi and also with various strings. The behavior on all of them is the same.
For example, call  `Serialize("Hello")` produces the following output:
```
8kjNyckHAAAA%2F%2F8DAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA%3D%3D
```
And further investigation, the culprit seems to be the `compressed.GetBuffer()` call. Replacing `GetBuffer()` with `ToArray()` seems to fix the problem. Call `Serialize("Hello")` produces then:
```
8kjNyckHAAAA%2F%2F8DAA%3D%3D
```

And that output seems correct. My program is running on dotnet 7.0 so perhaps it's simply something caused by the version. Do you have any ideas what might be causing the problem and do you think changing the `GetBuffer()` call to `ToArray()` is a sufficient workaround?

**Comment by AlexEngblom** [comment]
Solution seems to be correct. According to MSDN, buffer may very likely have some allocated but unused bytes.
https://learn.microsoft.com/en-us/dotnet/api/system.io.memorystream.getbuffer?view=net-7.0&redirectedfrom=MSDN#System_IO_MemoryStream_GetBuffer

The ToArray() returns just the actual data which we are interested in. Only downside is that it creates a copy of the data in memory, but this isn't probably of any issue in our case.

Seems there is a fix pending for this, which I'll check into now:
#9

**Comment by AlexEngblom** [comment]
Fixed with #9 and published in 3.0.4 NuGet.
